### PR TITLE
Readline support with pandas additions from PR #18

### PR DESCRIPTION
**Pull request by martindurant**
Special attention to rerely-used length parameter

### CODE REVIEW

File: s3fs/core.py [hunk 1]
```diff
@@ -545,6 +545,35 @@ def seek(self, loc, whence=0):
...
+        '''
+        Read and return a line from the stream.
+        
+        If size is specified, at most size bytes will be read.
+        '''  
+        self._fetch(self.loc, self.loc+1)
+        while True:
+            found = self.cache[self.loc-self.start:].find(b'\n') + 1
+            if length > 0 and found > length:
+                return self.read(length)
```
Comment (TomAugspurger) [inline s3fs/core.py]: Much simpler to reuse the logic in `.read`, nice!

File: s3fs/core.py [hunk 2]
```diff
@@ -545,6 +545,35 @@ def seek(self, loc, whence=0):
         self.loc = nloc
         return self.loc
 
+    def readline(self, length=-1):
+        '''
+        Read and return a line from the stream.
+        
+        If size is specified, at most size bytes will be read.
```
Comment (TomAugspurger) [inline s3fs/core.py]: Maybe want to change `size` to `length`. I copied that from the `io` docs, which use `size` for the positional argument name, but s3fs uses `length`.

File: s3fs/core.py [hunk 3]
```diff
@@ -652,6 +681,18 @@ def close(self):
                 self.s3.s3.put_object(Bucket=self.bucket, Key=self.key)
             self.s3._ls(self.bucket, refresh=True)
 
+    def readable(self):
```
Comment (jcrist) [inline s3fs/core.py]: Would these be better as properties?

Comment (TomAugspurger) [inline s3fs/core.py]: They have to be callable to be compatible with the `io` base classes (which surprised me that they weren't properties).

**Pull request merged**